docs/migration.md

```diff
@@ -0,0 +1,549 @@
+  "podPort": 443,
+  "agentHost": "agent.symphony.com",
+  "agentPort": 443,
+  "botPrivateKeyPath": "certs/",
+  "botPrivateKeyName": "/path/to/bot/rsa-privatekey.pem",
+  "botUsername": "bot-username",
+  "authTokenRefreshPeriod": "30",
+  "authenticationFilterUrlPattern": "/secure/",
+  "showFirehoseErrors": false,
+  "connectionTimeout": 45000, 
+  "proxyURL": "proxy.symphony.com",
+  "proxyUsername": "proxy.username",
+  "proxyPassword": "proxy.password",
+  "keyManagerProxyURL": "km.proxy.symphony.com",
+  "keyManagerProxyUsername": "km.proxy.username",
+  "keyManagerProxyPassword": "km.proxy.password"
+}
+```
+</td>
+<td>
+
+Only `config.yaml` file is required. It can be in *JSON* 
+
+
+```json
+{
+    "bdk": {
+        "host": "acme.symphony.com",
+        "bot": {
+            "username": "bot-username",
+            "privateKey": {
+                "path": "/path/to/bot/rsa-privatekey.pem"
+            }
+        },
+        "proxy": {
+            "host": "proxy.symphony.com",
+            "port": 1234,
+            "username": "proxyuser",
+            "password": "proxypassword"
+        },
+        "keyManager": {
+            "proxy": {
+                "host": "km.proxy.symphony.com",
+                "port": 1234,
+                "username": "km.proxy.username",
+                "password": "km.proxy.password"
+            }
+        }   
+    }
+}
+```
+
+or *YAML* format.
+
+```yaml
+bdk:
+    host: acme.symphony.com
+    bot:
+        username: bot-username
+        privateKey:
+            path: "/path/to/bot/rsa-privatekey.pem"
+    proxy:
+        host: proxy.symphony.com
+        port: 1234
+        username: proxyuser
+        password: proxypassword
+    keyManager:
+        proxy:
+            host: km.proxy.symphony.com
+            port: 1234
+            username: km.proxy.username
+            password: km.proxy.passwor
+```
+</td>
+</tr>
+</table>
+
+#### Non framework based project
+<table>
+<tr>
+<th>Java BDK 1.0</th>
+<th>Java BDK 2.0</th>
+</tr>
+<tr>
+<td>
+
+#### **`bot-config.json`:**
+```json
+{
+  "sessionAuthHost": "session.symphony.com",
+  "sessionAuthPort": 443,
+  "keyAuthHost": "km.symphony.com",
+  "keyAuthPort": 443,
+  "podHost": "pod.symphony.com",
+  "podPort": 443,
+  "agentHost": "agent.symphony.com",
+  "agentPort": 443,
+  "botPrivateKeyPath": "certs/",
+  "botPrivateKeyName": "/path/to/bot/rsa-privatekey.pem",
+  "botUsername": "bot-username",
+  "authTokenRefreshPeriod": "30",
+  "authenticationFilterUrlPattern": "/secure/",
+  "showFirehoseErrors": false,
+  "connectionTimeout": 45000, 
+  "proxyURL": "proxy.symphony.com",
+  "proxyUsername": "proxy.username",
+  "proxyPassword": "proxy.password",
+  "keyManagerProxyURL": "km.proxy.symphony.com",
+  "keyManagerProxyUsername": "km.proxy.username",
+  "keyManagerProxyPassword": "km.proxy.password"
+}
+```
+</td>
+<td>
+
+Only `config.yaml` file is required. It can be in *JSON*
+
+
+```json
+{
+    "bdk": {
```

Comment:
For non spring boot project, there is not `bdk` key (the `bdk` key only applies for the spring boot based projects)